### PR TITLE
cleanup + refactoring of target platform and maven build xpect_latest

### PR DESCRIPTION
**Pull request by mehmet-karaman**
Refactored the target platform and adjusted the maven build according to that.

### CODE REVIEW

**Comment by cdietrich** [comment]
As you are editing the Jenkins file am not sure if it will be properly picked

If not can you directly push to the eclipse repo as branch

**Comment by mehmet-karaman** [comment]
There are heavy changes for my knowledge :) I build both succesfully the mvn build with the two different profiles.. and it worked.. So I hope we could detect any other problem here.. If I push it directly to the xpect repo, can we be sure that everything is alright?

**Comment by cdietrich** [comment]
Yes would be helpful,to have branch directly in repo just to be sure

File: org.eclipse.xpect/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -22,7 +22,7 @@ Require-Bundle: org.eclipse.xtext,
  org.eclipse.xtext.common.types;visibility:=reexport,
  org.apache.log4j;bundle-version="1.2.0";visibility:=reexport,
  org.objectweb.asm;bundle-version="[9.5.0,10.0.0)";resolution:=optional
-Bundle-RequiredExecutionEnvironment: JavaSE-17
+Bundle-RequiredExecutionEnvironment: JavaSE-11
```
Comment (cdietrich) [inline org.eclipse.xpect/META-INF/MANIFEST.MF:25]: Intentional ?

Comment (mehmet-karaman) [reply org.eclipse.xpect/META-INF/MANIFEST.MF:25]: wrong merge/rebase conflict resolving

**Review by cdietrich — APPROVED** [review]

**Comment by cdietrich** [comment]
Thx @mehmet-karaman

**Comment by mehmet-karaman** [comment]
#365 preparations for adjusting the mwe, xtext and eclipse platform is done with this PR

**Pull request merged**